src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/AddRefTests.cs

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Runtime.InteropServices.Tests
+{
+    public class AddRefTests
+    {
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void AddRef_ValidPointer_Success()
+        {
+            IntPtr iUnknown = Marshal.GetIUnknownForObject(new object());
+            Assert.Equal(2, Marshal.AddRef(iUnknown));
+            Assert.Equal(3, Marshal.AddRef(iUnknown));
+
+            Marshal.Release(iUnknown);
+            Marshal.Release(iUnknown);
+            Assert.Equal(2, Marshal.AddRef(iUnknown));
```

Comment:
Nit: Could you release all of these AddRef?

Reply:
Done